File: packages/agentmark-core/src/adapters/default.ts
import { 
  Adapter, 
  TextConfig,
  ImageConfig,
  ObjectConfig, 
  PromptShape
} from "../types";

export class DefaultAdapter<
  T extends PromptShape<T> = any
> implements Adapter<T> {
  declare readonly __dict: T;
  

  adaptText(
    input: TextConfig,
  ): TextConfig {
    return input;
  }

  adaptObject(
    input: ObjectConfig,
  ): ObjectConfig {
    return input;
  }

  adaptImage(
    input: ImageConfig,
  ): ImageConfig {
    return input;
  }

  adaptSpeech(
    input: any,
  ): any {
    return input;
  }
}

File: packages/prompt-core-python/src/agentmark/prompt_core/adapters/default.py
"""Default passthrough adapter."""

from ..schemas import (
    ImageConfigSchema,
    ObjectConfigSchema,
    SpeechConfigSchema,
    TextConfigSchema,
)
from ..types import AdaptOptions, PromptMetadata


class DefaultAdapter:
    """Default passthrough adapter that returns configs unchanged."""

    @property
    def name(self) -> str:
        """Adapter name identifier."""
        return "default"

    def adapt_text(
        self,
        config: TextConfigSchema,
        _options: AdaptOptions,
        _metadata: PromptMetadata,
    ) -> TextConfigSchema:
        """Return text config unchanged."""
        return config

    def adapt_object(
        self,
        config: ObjectConfigSchema,
        _options: AdaptOptions,
        _metadata: PromptMetadata,
    ) -> ObjectConfigSchema:
        """Return object config unchanged."""
        return config

    def adapt_image(
        self,
        config: ImageConfigSchema,
        _options: AdaptOptions,
    ) -> ImageConfigSchema:
        """Return image config unchanged."""
        return config

    def adapt_speech(
        self,
        config: SpeechConfigSchema,
        _options: AdaptOptions,
    ) -> SpeechConfigSchema:
        """Return speech config unchanged."""
        return config
